### PR TITLE
use of a single collector to scrape 1337x target

### PR DESCRIPTION
**Pull request by ash-kamrip**
#42 I have now implemented the use of a single collector now, Let me know if you have any questions regarding it.

### CODE REVIEW

File: scraper/target1337x.go [hunk 1]
```diff
@@ -4,62 +4,96 @@ import (
...
-		newMovie.Magnet = magnet
-		results = append(results, newMovie)
+		metadata := h.DOM
 
+		movieName := metadata.Find("div.box-info-heading h1").Text()
+		metadata.Find("a").Each(func(i int, s *goquery.Selection) {
+			if magnet, exists := s.Attr("href"); exists {
+				if strings.Contains(magnet, "magnet") {
+					for i, movie := range movielist {
+						if compareNamesfromdifferentPages(strings.TrimSpace(movie.Name), strings.TrimSpace(movieName)) {
```
Comment (qascade) [inline scraper/target1337x.go:46]: rename  `compareNamesfromdifferentPages` -> `compareNamesFromDiffPages`. We need to maintain proper Camel Casing.

File: scraper/target1337x.go [hunk 2]
```diff
@@ -4,62 +4,96 @@ import (
...
 
-	s.collector.OnHTML("a", func(e *colly.HTMLElement) {
-		dom := e.DOM
-		potMagnet, _ := dom.Attr("href")
-		if strings.Contains(potMagnet, "magnet") {
-			magnet = potMagnet
+	if len(a) == 0 || len(b) == 0 {
+		return false
+	}
+	limit := len(a)
```
Comment (qascade) [inline scraper/target1337x.go:82]: What is the use of `limit` here?

File: scraper/target1337x.go [hunk 3]
```diff
@@ -4,62 +4,96 @@ import (
...
+
 	s.collector.OnHTML("tr", func(e *colly.HTMLElement) {
-		movieMetaDataDOM := e.DOM
-		anchorTagDom := movieMetaDataDOM.Find("a").Siblings()
 
-		var magnet string
-		if magnetPageUrl, exists := anchorTagDom.Attr("href"); exists {
-			magnet = getMagnetLink("https://1337x.to" + magnetPageUrl)
+		doc := e.DOM
+		anchortags := doc.Find("a").Siblings()
```
Comment (qascade) [inline scraper/target1337x.go:25]: rename `anchortags` -> `anchorTags`

File: scraper/target1337x.go [hunk 4]
```diff
@@ -4,62 +4,96 @@ import (
...
 
 	colly "github.com/gocolly/colly"
 )
 
+// scrape1337x returns the scraped results in Result type
 func (s *Scraper) scrape1337x(context *QueryContext) (results []Result, err error) {
 
+	var links []string
+	var movielist []movie.Movie
 	results = make([]Result, 0)
```
Comment (qascade) [inline scraper/target1337x.go:18]: make `var` declaration convention of slices uniform. Change `results = make([]Result,0)` -> `var results []Result`

**Review by qascade — CHANGES_REQUESTED** [review]
Please move the `ISSUE_TEMPLATE.md` and `PULL_REQUEST_TEMPLATE.md` to `.workflows` directory.

**Comment by qascade** [comment]
@ash-kamrip  I have tested the changes locally, the code is working fine and the stream is starting without any issues. Other than minor variable name changes, we are good to go

File: scraper/target1337x.go (hunk 1 above)
Comment (qascade) [reply scraper/target1337x.go:46]: Also, can you explain why do we need this function?

Comment (ash-kamrip) [reply scraper/target1337x.go:46]: movie struct is populated from "tr" (table rows) , i compare the name of the movie from page 1 to page 2 in order to insert the magnet later when found , but the problem i faced was on page 2 ,1337x limits the name of the movie till 65 characters and append `...` to it. Hence created a function to compare movie name from different webpages

File: scraper/target1337x.go (hunk 2 above)
Comment (ash-kamrip) [reply scraper/target1337x.go:82]: as previously explained limit used when the movie name is larger than 65 characters. we run the loop for only first 65 characters , as 1337x limits the name till this limit

File: scraper/target1337x.go (hunk 3 above)
Comment (ash-kamrip) [reply scraper/target1337x.go:25]: yes

File: scraper/target1337x.go (hunk 4 above)
Comment (ash-kamrip) [reply scraper/target1337x.go:18]: yes ok

**Comment by ash-kamrip** [comment]
One more structural change i think is we should not have a interface for collector object , but then since we taken input a collector alredy was a little confused as to how to achieve it. I wanted to code this with parallel threads but that is something we can think about later i guess.

File: scraper/target1337x.go (hunk 2 above)
Comment (ash-kamrip) [reply scraper/target1337x.go:82]: to better explain this , imagine the movie name is of 70 characters on page 1 ,  on page 2 the movie name will be truncated to 68 characters i.e. 65 characters of movie name + `...`  , so limit takes the smaller of (70,68) if the val is still larger than 65 which it always is gonna be , we will only loop through 65 chars.

**Comment by ash-kamrip** [comment]
@qascade required changes are done now.

File: scraper/target1337x.go (hunk 2 above)
Comment (qascade) [reply scraper/target1337x.go:82]: Ok, So I can consider this function as an ad hoc implementation only relevant to 1337x then. Makes sense.

**Comment by qascade** [comment]
> 

Collector is a part of the `Scraper` struct which is not an interface. Collector in itself is whether an interface or not is not our concern as it is a design decision of Go-Colly peeps. If you want to work to further optimize this using concurrency. You are free to do that. We can do that in this PR itself. Other than that Looks good to merge. If you are planning to work on further optimization, do let me know otherwise I will merge the PR. Thanks for your contribution.

**Review by qascade — APPROVED** [review]
Looks good to merge. I left a comment regarding your suggestion to optimize query searches using concurrency.

**Comment by ash-kamrip** [comment]
I am browsing other repos for hacktoberfest as of now , would not be able to optimize this further at the moment. Kindly merge this PR as of now.

**Pull request merged**